chain/rosetta-rpc/src/adapters/mod.rs

```diff
@@ -144,54 +142,66 @@ pub(crate) async fn convert_block_to_transactions(
         .await?
         .map_err(crate::errors::ErrorKind::InternalError)?;
 
-    let mut transactions = Vec::<crate::models::Transaction>::new();
-    for account_change in accounts_changes.into_iter() {
+    let transactions = convert_block_changes_to_transactions(
+        &genesis.config.runtime_config,
+        &block.header.hash,
+        accounts_changes,
+        accounts_previous_state,
+    )?;
+    Ok(transactions.into_iter().map(|(_transaction_hash, transaction)| transaction).collect())
+}
+
+fn convert_block_changes_to_transactions(
```

Comment:
I refactored it into a separate testable function (and covered with snapshot tests)